server/middleware/measure.go

```diff
@@ -117,37 +117,51 @@ func measureStream() grpc.StreamServerInterceptor {
 }
 
 func (w *wrappedStream) RecvMsg(m interface{}) error {
+	err := w.ServerStream.RecvMsg(m)
+	if err != nil {
+		ulog.E(err)
+	}
 	if w.measurement == nil {
-		err := w.ServerStream.RecvMsg(m)
 		return err
 	}
-	db, coll := request.GetDbAndColl(m)
-	reqMetadata, err := request.GetRequestMetadataFromContext(w.WrappedContext)
-	if err != nil {
-		return err
+
+	if len(w.measurement.GetDBCollTags()) == 0 {
+		// The request is not tagged yet with db and collection, need to do it on the first message
+		db, coll := request.GetDbAndColl(m)
+		reqMetadata, err := request.GetRequestMetadataFromContext(w.WrappedContext)
+		if err != nil {
+			ulog.E(err)
+		}
+		reqMetadata.SetDb(db)
+		reqMetadata.SetCollection(coll)
+		w.measurement.AddDbCollTags(db, coll)
 	}
-	reqMetadata.SetDb(db)
-	reqMetadata.SetCollection(coll)
 
-	err = w.ServerStream.RecvMsg(m)
-	dbCollTags := metrics.GetDbCollTags(reqMetadata.GetDb(), reqMetadata.GetCollection())
-	w.measurement.RecursiveAddTags(dbCollTags)
 	w.measurement.CountReceivedBytes(metrics.BytesReceived, w.measurement.GetNetworkTags(), proto.Size(m.(proto.Message)))
 	return err
 }
 
 func (w *wrappedStream) SendMsg(m interface{}) error {
+	err := w.ServerStream.SendMsg(m)
+	if err != nil {
+		return errors.Internal("Could not handle stream send message")
+	}
 	if w.measurement == nil {
-		err := w.ServerStream.SendMsg(m)
 		return err
 	}
-	err := w.ServerStream.SendMsg(m)
-	reqMetadata, err1 := request.GetRequestMetadataFromContext(w.WrappedContext)
-	if err1 != nil {
-		return errors.Internal("Could not handle stream send message")
+
+	if len(w.measurement.GetDBCollTags()) == 0 {
+		// The request is not tagged yet with db and collection, need to do it on the first message
+		db, coll := request.GetDbAndColl(m)
+		reqMetadata, err := request.GetRequestMetadataFromContext(w.WrappedContext)
+		if err != nil {
+			ulog.E(err)
```

Comment:
Could this potentially blow up our logging if it gets triggered on `SendMsg()`?

Reply:
Perhaps we could set the error flag on measurement to avoid logging N times?

Reply:
Yes, this is for each document in the result set. One may have error, the other may not. We can also just log it at debug level or something like that.